src/uint/add.rs

```diff
@@ -99,6 +101,49 @@ impl<const LIMBS: usize> WrappingAdd for Uint<LIMBS> {
     }
 }
 
+/// Two argument addition of raw slices:
+/// a += b
+///
+/// The caller _must_ ensure that a is big enough to store the result - typically this means
+/// resizing a to max(a.len(), b.len()) + 1, to fit a possible carry.
+pub(crate) fn add2(a: &mut [Limb], b: &[Limb]) {
+    let carry = __add2(a, b);
+
+    assert!(num_traits::Zero::is_zero(&carry));
+}
+
+#[inline]
+fn __add2(a: &mut [Limb], b: &[Limb]) -> Limb {
+    debug_assert!(a.len() >= b.len(), "{} < {}", a.len(), b.len());
+
+    let mut carry = 0;
+    let (a_lo, a_hi) = a.split_at_mut(b.len());
+
+    for (a, b) in a_lo.iter_mut().zip(b) {
+        *a = Limb(adc(a.0, b.0, &mut carry));
+    }
+
+    if carry != 0 {
+        for a in a_hi {
+            *a = Limb(adc(a.0, 0, &mut carry));
+            if carry == 0 {
+                break;
+            }
+        }
+    }
+
+    Limb(carry as Word)
+}
+
+#[inline]
+pub(crate) fn adc(a: Word, b: Word, acc: &mut WideWord) -> Word {
+    *acc += a as WideWord;
+    *acc += b as WideWord;
+    let lo = *acc as Word;
+    *acc >>= Word::BITS;
+    lo
+}
```

Comment:
`Limb` has an `adc` method you can call directly.

Also this whole module feels like implementation details of Karatsuba. Maybe move it to `karatsuba.rs`?

Reply:
I am using parts of this in the `div` impl, that's why I moved it into the shared piece.